internal/resource/resourcetest/builder.go

```diff
@@ -27,7 +28,12 @@ type resourceBuilder struct {
 	dontCleanup bool
 }
 
-func Resource(rtype *pbresource.Type, name string) *resourceBuilder {
+func Resource(rtype *pbresource.Type, name string, registry resource.Registry) *resourceBuilder {
```

Comment:
The ergonomics of this are going to be annoying in practice. I wonder if we could make something like `resourcetest.New(registry) *Helper` that has a method like `func (h *Helper) Resource(rtype *pbresource.Type, name string) *resourceBuilder` and you could use that throughout a unit test and only thread the `registry` through one call site instead of dozens.